src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6623,7 +6627,7 @@ public override IOperation VisitInvalid(IInvalidOperation operation, int? captur
             IOperation visitInvalidOperationStatement(IInvalidOperation invalidOperation)
             {
                 Debug.Assert(_currentStatement == invalidOperation);
-                VisitStatements(children);
+                VisitStatements(children.ToImmutable());
```

Comment:
>ToImmutable [](start = 41, length = 11)

ToImmutableAndFree and adjust other places to do the same?

Reply:
Do we want to actually free the children though? They aren't owned by the walker so must be freed somewhere else?

Reply:
> Do we want to actually free the children though? They aren't owned by the walker so must be freed somewhere else?

We are in a local function and ```children``` is a builder created by the containing method. It is freed before we return from the method. It looks like instead of calling ```Free``` before the return, we could call ```ToImmutableAndFree``` here and in ```visitInvalidOperationExpression```, replacing ```invalidOperation.Children.ToImmutableArray()``` with ```children.ToImmutableAndFree()```.  

---
In reply to: [333732853](https://github.com/dotnet/roslyn/pull/39125#discussion_r333732853) [](ancestors = 333732853)